x-pack/plugins/index_management/__jest__/client_integration/helpers/http_requests.ts

```diff
@@ -11,145 +11,93 @@ import { API_BASE_PATH } from '../../../common/constants';
 type HttpResponse = Record<string, any> | any[];
 
 export interface ResponseError {
-  statusCode: number;
-  message: string | Error;
-  attributes?: Record<string, any>;
+  status: number;
+  error: string;
+  message: string;
 }
 
 // Register helpers to mock HTTP Requests
-const registerHttpRequestMockHelpers = (server: SinonFakeServer) => {
-  const setLoadTemplatesResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/index_templates`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadIndicesResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/indices`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setReloadIndicesResponse = (response: HttpResponse = []) => {
-    server.respondWith('POST', `${API_BASE_PATH}/indices/reload`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadDataStreamsResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/data_streams`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
+export const init = () => {
+  const server = sinon.fakeServer.create();
```

Comment:
Slightly unrelated, but there's [another pr](https://github.com/elastic/kibana/pull/127122) that is trying to fix the incorrect usage of axios+sinon that we have throughout our apps in order to use `HttpSetup` instead. There's a few very nice abstractions that are created there that I can see you also thought about here! I'll create an issue to patch this up later on when the other PR is merged